### PR TITLE
[pyspark] Support collective.Conf in spark

### PR DESCRIPTION
**Pull request by wbo4958**
This PR is a followup of https://github.com/dmlc/xgboost/pull/10983

### CODE REVIEW

File: python-package/xgboost/spark/core.py [hunk 1]
```diff
@@ -257,21 +258,20 @@ class _SparkXGBParams(
         "launched on the driver side; otherwise, it will be launched on the executor side.",
         TypeConverters.toBoolean,
     )
-    tracker_host_ip = Param(
+    tracker = Param(
```
Comment (wbo4958) [inline python-package/xgboost/spark/core.py:261]: the name "tracker" is ok?

Comment (trivialfis) [reply python-package/xgboost/spark/core.py:261]: I would probably continue to use the name `coll_cfg` just to be consistent. Besides, the configuration is for both the tracker and the communicators.

Comment (wbo4958) [reply python-package/xgboost/spark/core.py]: sure, I changed it to `collective_conf`

Comment (trivialfis) [reply python-package/xgboost/spark/core.py]: I know everyone has a naming preference ...  if you use a new name, please modify the existing ones in Dask for consistency.

Comment (wbo4958) [reply python-package/xgboost/spark/core.py]: OK, I just changed it to coll_cfg to be compatible with dask.

**Review by trivialfis — APPROVED** [review]
Thank you for the PR!

**Pull request merged**